### PR TITLE
Implement class GuitarCourseState

### PR DESCRIPTION
**Pull request by nikitochka-akaka**
Classes: Added
-	GuitarCourseState abstract class (defines current state of guitar courses)
-	Six inheritors of the "GuitarCourseState" class (each one defines a specific current state)

Class StudyCourseState:
-	There were several bugs in this superclass that were very annoying:
    -	Abstract methods had bodies (Fixed)
    -	There was incorrect parameter in 'buyNextCourse' method (Fixed)

### CODE REVIEW

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package android.clicker.school_live_simulator
+
+abstract class GuitarCourseState: StudyCourseState() {
+    abstract override val price: Int
```
Comment (Bazalii) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt:4]: do we need override here?

**Review by Bazalii — COMMENTED** [review]
I have only one question and also we agreed to use present tenses in commits(not added but add but it's a recommendation for future don;t change anything now)

**Review by Bazalii — APPROVED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt (hunk 1 above)
Comment (Ferrociraptus) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt:4]: I think that it is not necessary. 
@nikitochka-akaka can you remove overriding this field from this abstract class?

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     * @param   song    name of the song
+     * @return          song availability (true/false)
+     */
+    open fun isAvailable(song: Song): Boolean {
+        return available_playlist.contains(song)
+    }
+
+    abstract override fun timerTickHandler()
+
+    abstract override fun timerEndHandler()
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt:25]: All abstract fields shouldn't be override in abstract children classes. 
You should remove all abstract field overriding.

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt (hunk 2 above)
Comment (Bazalii) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt:25]: And also please add prices from firstsongcourse and they will be 100, 300, 900, 2700, 8100

**Review by Bazalii — CHANGES_REQUESTED** [review]
And also please add prices from firstsongcourse. Let them be 100, 300, 900, 2700

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt (hunk 2 above)
Comment (nikitochka-akaka) [reply Project_source/app/src/main/java/android/clicker/school_live_simulator/GuitarCourseState.kt]: Fixed

**Review by Bazalii — APPROVED** [review]

File: Project_source/app/src/main/java/android/clicker/school_live_simulator/MusicalObservatoryCourseState.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    override fun buyNextCourse(courses: Player.Courses):
+        Unit = throw UnsupportedOperationException("Operation not supported")
+
+    override fun timerTickHandler() {
+
+    }
+
+    override fun timerEndHandler() {
+
```
Comment (Ferrociraptus) [inline Project_source/app/src/main/java/android/clicker/school_live_simulator/MusicalObservatoryCourseState.kt:14]: Please, throw exception that it is last course as it do @nikitochka-akaka.

**Review by Ferrociraptus — CHANGES_REQUESTED** [review]

**Review by Ferrociraptus — APPROVED** [review]

**Pull request merged**